org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -200,7 +189,8 @@ private boolean checkIfReady() {
       .collect(Collectors.toSet());
 
     for (var configurationScopeId : configurationScopeIds) {
-      if (Boolean.TRUE.equals(analysisReadyByConfigurationScopeId.get(configurationScopeId))) {
+      var readyness = AnalysisReadyStatusCache.getAnalysisReadiness(configurationScopeId);
+      if (Boolean.TRUE.equals(readyness)) {
```

Comment:
```suggestion
      if (readyness) {
```

Reply:
Thanks for the comment, due to this I found a bug here :smile: We were only waiting for one project to be ready and not for all of them...